shop/src/test/java/ro/msg/learning/shop/LocationStrategyTests.java

```diff
@@ -0,0 +1,143 @@
+package ro.msg.learning.shop;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.test.context.SpringBootTest;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.domain.Stock;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.dto.ProductIdAndQuantity;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.service.StockService;
+import ro.msg.learning.shop.strategies.MostAbundantStrategy;
+import ro.msg.learning.shop.strategies.SingleLocationStrategy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+class LocationStrategyTests {
+
+    @Mock
+    private StockService stockService;
+
+    @Mock
+    private ProductService productService;
+
+    @InjectMocks
+    private MostAbundantStrategy mostAbundantStrategy;
+
+    @InjectMocks
+    private SingleLocationStrategy singleLocationStrategy;
+
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void testSingleLocationStrategy_ValidOrder() {
```

Comment:
1. firstObject and secondObject are not names. give them names as expetected because i understood nothing from the test
2. pupulating uuids and locations and other things can happen a) in a separate private method in this class b) in a before or beforeEach method c) in a public method from a dataSource for testing class. But not all here
3. The test is confusing a bit. Take into consideration what i wrote and try and refactor it to become more comprehensible. When someone else from an entirely different project looks at it they should be able to tell what happens just by seeing the code. 
4. use comments for defining "given, when, then"
5. if it tests more things, make more tests